### ISSUE TITLE
Error Running Sample Execution in Da Vinci DTR Payer Server Test Suite in Local Instance

### ISSUE THREAD

**Issue by j2cj2c** [inferno-framework/davinci-dtr-test-kit#28]
In the [Da Vinci DTR Payer Server Test Suite](https://inferno.healthit.gov/test-kits/davinci-dtr/), underneath **Running the Tests** section, there is a section about **Sample Execution**.
![image](https://github.com/user-attachments/assets/ba6df812-8eca-4b7b-bdf6-6a47e0131119)

In the website, **Sample Execution** runs without an issue.

In the local instance, I am having trouble replicating the result.

I am running docker inside of an Ubuntu running on top of WSL.

Here are the steps that I am taking to run the test kits locally.

1. I clone the source repository.
2. I build and run the docker containers using setup.sh and run.sh.
3. I connect to http://localhost
4. I follow the steps 1 - 5 in the **Sample Execution**. The resulting popup instructs to send the question package request to a base URL of http://localhost:4567/custom/dtr_smart_app/fhir
5. I follow the steps 6 - 10 in the **Sample Execution**. 
6. Test 1.03 of **Static Questionnaire Package Retrieval** fails with an error message that states that it could not establish a TCP connection to http://localhost:4567.
![image](https://github.com/user-attachments/assets/e861163d-2b01-4eed-b83d-8de695f56f42)

Is **Sample Execution** only supposed to be used for the web instance? If not, what is the base URL that I should be using for test 1 _Static Questionnaire Package Retrieval_ when running a local instance?

**Comment by arscan** [comment]
Hi @j2cj2c --

Thanks for the question. The problem here is that when running in a pure dockerized environment (using `./start.sh` and `./run.sh`), the container that is running the tests is separate from the container that is hosting a server to connect to, therefore "localhost" is not the same.  The reason why it works on inferno.healthit.gov is that "inferno.healthit.gov" resolves back to the same spot properly regardless of the container it is being resolved from, which isn't the case for 'localhost'.

I'm not sure about Windows with WSL, but at least on Macs, Docker has a special host `host.docker.internal` that resolves back up to the host's "localhost".  So if you use that instead of "localhost" it should work (or at least, it does when on a Mac).  

<img width="994" alt="Screenshot 2024-11-14 at 4 36 55 PM" src="https://github.com/user-attachments/assets/2f2e3c3e-d0f3-4d9a-a6f4-a253fe3b9b44">



Also, one minor configuration error it looks like: when running in a pure docker environment, you want the .env file to have this line:

```
INFERNO_HOST="http://localhost"
```
The link in the "wait dialog" at the bottom to continue the smart app tests will point to "http://localhost:4567" otherwise, which isn't correct. The "4567" should only be used when running in developer mode, which runs the application within Ruby, which is what our team usually uses.

<img width="599" alt="Screenshot 2024-11-14 at 4 33 44 PM" src="https://github.com/user-attachments/assets/70a24d5d-af98-465b-a66b-41ad277f0067">


An alternative if the `host.docker.internal` doesn't work is you could try running this all in the 'developer mode' with Ruby installed with [these instructions ](https://inferno-framework.github.io/docs/getting-started/#run-your-test-kit), which will have all components running in Ruby on your machine, but that involves installing Ruby which isn't quite as easy as just using Docker.

Let me know if this helps!

Thanks,
Rob

**Comment by j2cj2c** [comment]
Thank you, @arscan! That did allow for Payer Server Test Suite to fetch questions from SMART App Test Suite.
I did have to add a configuration to make Nginx to listen on port 8080 instead because Windows had a default service listening on port 80, but, other than that, I made the changes as you said.

Here's the screenshot of the result that I got for the _1 Static Questionnaire Package Retrieval_.
![image](https://github.com/user-attachments/assets/a17737f1-2a55-4da6-a02b-7a1383a0cc97)
![image](https://github.com/user-attachments/assets/1fa31141-e637-4e77-8df3-94c9674034ac)
![image](https://github.com/user-attachments/assets/c100a838-0285-4657-8b31-1b9194ebe9a8)

Has there been an update in the implementation lately? Just curious as to why the test cases are failing.
